lasagne/layers/special.py

```diff
@@ -354,3 +356,63 @@ def _meshgrid(height, width):
     ones = T.ones_like(x_t_flat)
     grid = T.concatenate([x_t_flat, y_t_flat, ones], axis=0)
     return grid
+
+
+class ParametricRectifierLayer(Layer):
+    """
+    lasagne.layers.ParametricRectifierLayer(incoming,
+    alpha=init.Uniform((0.001, 0.1)), **kwargs)
+
+    A layer that applies parametric rectify nonlinearity to its input
+    following K He, X Zhang et al. (http://arxiv.org/abs/1502.01852):
+    :math:`\\varphi(y_i) = \\max(x_i,0) + \\alpha_i \\min(x_i,0)`
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape
+
+    a : Theano shared variable, numpy array or callable
+        An initializer for the weights of the layer. If a shared variable or a
+        numpy array is provided the shape should  be incoming.shape[1:]).
+        See :meth:`Layer.create_param` for more information.
+    """
+    def __init__(self, incoming, alpha=init.Uniform((0.001, 0.1)), axes=None,
+                 **kwargs):
+        super(ParametricRectifierLayer, self).__init__(incoming, **kwargs)
+        if axes is None:
+            # default: the parameters are same across only second axis
+            # this will do correct thing for both dense operations
+            # and convolution
+            axes = (1, )
+        elif isinstance(axes, int):
+            axes = (axes,)
+        shape = [1, ] * len(self.input_shape)
+        for axis in axes:
+            shape[axis] = self.input_shape[axis]
+        self.broadcast = tuple(set(range(len(self.input_shape))) - set(axes))
+        if any(size is None for size in shape):
+            raise ValueError("ParametricRectifierLayer needs specified input "
+                             "sizes for all dimensions/axes which constitute "
+                             "the same parameters.")
```

Comment:
I don't understand this sentence (the last part, actually: "all dimensions/axes which constitute the same parameters").